kotlin_intro/src/test/kotlin/com/infinum/homework/service/CarCheckUpServiceUnitTests.kt

```diff
@@ -0,0 +1,132 @@
+package com.infinum.homework.service
+
+import com.infinum.homework.ApplicationConfiguration
+import com.infinum.homework.exception.NoVinException
+import com.infinum.homework.model.Car
+import com.infinum.homework.model.CarCheckUp
+import com.infinum.homework.repository.CarCheckUpRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.time.LocalDateTime
+import java.time.Month
+import kotlin.random.Random
+
+@ExtendWith(SpringExtension::class)
+@ContextConfiguration(classes = [ApplicationConfiguration::class])
+class CarCheckUpServiceUnitTests {
+    private lateinit var service: CarCheckUpService
+    private val repository = mockk<CarCheckUpRepository>()
+    private val cars: List<Car> = listOf(
+        Car("BMW", "M3", "JH4DA9470MS008988"),
+        Car("Audi", "A4", "JH4KA3171KC016159"),
+        Car("Mercedes", "C200", "5NPDH4AE0DH213924"),
+        Car("Citroen", "C4", "JHMBB2269NC026648"),
+    )
+
+    @BeforeEach
+    fun setUp() {
+        every { repository.insert(any(), any()) } returns Random.nextLong()
+        service = CarCheckUpService(repository)
+        cars.forEach { car -> service.addCar(car) }
+    }
+
+    @Test
+    fun isCheckUpNecessaryCarCheckedLessThanYearAgo() {
+        every { repository.findAll().values } returns listOf(
+            CarCheckUp(1, LocalDateTime.of(2022, Month.MAY, 25, 13, 30), cars[0])
+        )
+        assertThat(service.isCheckUpNecessary(cars[0].vin)).isFalse
+        verify(exactly = 1) { repository.findAll() }
+    }
+
+    @Test
+    fun isCheckUpNecessaryCarCheckedMoreThanYearAgo() {
+        every { repository.findAll().values } returns listOf(
+            CarCheckUp(1, LocalDateTime.of(2020, Month.MAY, 25, 13, 30), cars[0])
+        )
+        assertThat(service.isCheckUpNecessary(cars[0].vin)).isTrue
+        verify(exactly = 1) { repository.findAll() }
+    }
+
+    @Test
+    fun isCheckUpNecessaryCarNeverChecked() {
+        every { repository.findAll().values } returns listOf(
+            CarCheckUp(1, LocalDateTime.of(2020, Month.MAY, 25, 13, 30), cars[0])
+        )
+        assertThat(service.isCheckUpNecessary(cars[1].vin)).isTrue
+        verify(exactly = 1) { repository.findAll() }
+    }
+
+    @Test
+    fun isCheckUpNecessaryCarNotFound() {
+        every { repository.findAll().values } returns listOf(
+            CarCheckUp(1, LocalDateTime.of(2020, Month.MAY, 25, 13, 30), cars[0])
+        )
+        assertThat(service.isCheckUpNecessary("JH4DA9470MS008981")).isFalse
```

Comment:
Ovaj test faila jer ovdje moraš eksplicitno assertat da case baca exception:
```suggestion
 assertThatThrownBy { service.isCheckUpNecessary("JH4DA9470MS008981") }
            .isInstanceOf(NoVinException::class.java).hasMessage("Car with VIN JH4DA9470MS008981 does not exist")
```

Reply:
Ty 🙌